core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java

```diff
@@ -0,0 +1,332 @@
+package com.scalar.db.common;
+
+import com.scalar.db.api.DistributedTransactionAdmin;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.DataType;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public abstract class DecoratedDistributedTransactionAdmin implements DistributedTransactionAdmin {
+
+  private final DistributedTransactionAdmin decoratedDistributedTransactionAdmin;
```

Comment:
```suggestion
  private final DistributedTransactionAdmin distributedTransactionAdmin;
```
Should it be as above?
This class is a `DecoratedDistributedTransactionAdmin` so naming the member variable `decoratedDistributedTransactionAdmin` looks a bit weird to me.
What do you think?

Reply:
Ah, good point, thank you. Maybe the class name `DecoratedDistributedTransactionAdmin` is not good. I think `DistributedTransactionAdminDecorator` is a better name. What do you think?

Reply:
As discussed, we will rename the field name to `distributedTransactionAdmin`. Thanks.

Reply:
Fixed in 66dd2c920ae48795d471f7893b3dc79becd6513e. I've also renamed the field name in other `DecoratedXXX` classes.

@feeblefakie Could you please take a look again when you have time? Thanks.